paasta_tools/marathon_tools.py

```diff
@@ -307,13 +308,17 @@ def get_routing_constraints(self, service_namespace_config, system_paasta_config
         fake_slaves = [{"attributes": a} for a in expected_slave_attributes]
         filtered_slaves = filter_mesos_slaves_by_blacklist(
             slaves=fake_slaves,
-            blacklist=self.get_deploy_blacklist(
-                system_deploy_blacklist=system_paasta_config.get_deploy_blacklist(),
-            ),
-            whitelist=self.get_deploy_whitelist(
-                system_deploy_whitelist=system_paasta_config.get_deploy_whitelist(),
-            ),
+            blacklist=self.get_deploy_blacklist(),
+            whitelist=self.get_deploy_whitelist(),
+        )
+        # A slave must be allowed by both the instance config's blacklist/whitelist and the system configs' blacklist/
```

Comment:
strange, was this a bug just now being catched?

Reply:
Yup. The whitelist code previously was taking the system whitelist and the service's whitelist and concatenating them together, then unpacking them in `slave_passes_whitelist` with  `(location_type, locations) = whitelist`. This would fail if we ever set both the system whitelist and the service whitelist. Fortunately, we don't set the system whitelist.